src/libopensc/card-openpgp.c

```diff
@@ -2283,46 +2371,123 @@ pgp_calculate_and_store_fingerprint(sc_card_t *card, time_t ctime,
 
 	LOG_FUNC_CALLED(card->ctx);
 
-	/* temporary workaround: protect v3 cards against non-RSA */
-	if (key_info->algorithm != SC_OPENPGP_KEYALGO_RSA)
-		LOG_FUNC_RETURN(card->ctx, SC_ERROR_NOT_SUPPORTED);
+	/* constructing public-key packet length */
+	/* RSA */
+	if (key_info->algorithm == SC_OPENPGP_KEYALGO_RSA) {
 
-	if (modulus == NULL || exponent == NULL || mlen == 0 || elen == 0) {
-		sc_log(card->ctx, "Null data (modulus or exponent)");
-		LOG_FUNC_RETURN(card->ctx, SC_ERROR_INVALID_ARGUMENTS);
-	}
+		if (key_info->u.rsa.modulus == NULL 
+		   || key_info->u.rsa.exponent == NULL
+		   || (key_info->u.rsa.modulus_len >> 3) == 0
+		   || (key_info->u.rsa.exponent_len >> 3) == 0) {
```

Comment:
bad indent.

`key_info->u.rsa.modulus_len >> 3` looks strange. Don't you want to use some more readable comparison? Why are you enforcing these limits at all?

Reply:
This check was done before as well. Honestly, I am not totally sure if it is needed or not. I just hesitated to just delete it. The shifting is unnecessary and deleted now though.